### PR TITLE
feat: Zenseact Open Dataset (zod) support

### PR DESCRIPTION
**Pull request by egorchakov**
- add zod (https://zod.zenseact.com/)
- refactor TableAligner
- flatten batch structure
- more generic table processing: mapping -> pytree
- group sources/readers by filetype
- remove `scripts/build_table.py` + configs

### CODE REVIEW

File: config/_templates/dataset/carla.yaml [hunk 1]
```diff
@@ -59,14 +59,14 @@ _convert_: all
...
   (@=input_id@):
-    frame:
+    sources:
       #@ for source_id in cameras:
       (@=source_id@):
         index_column: _idx_
-        reader:
-          _target_: rbyte.io.frame.DirectoryFrameReader
+        sources:
+          _target_: rbyte.io.PathTensorSource
```
Comment (sandhawalia) [inline config/_templates/dataset/carla.yaml:67]: Nice abstraction

File: src/rbyte/io/video/ffmpeg_source.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from pydantic import FilePath, NonNegativeInt, validate_call
+from torch import Tensor
+from video_reader import (
+    PyVideoReader,  # pyright: ignore[reportAttributeAccessIssue, reportUnknownVariableType]
+)
+
+from rbyte.io.base import TensorSource
+
+
+class FfmpegFrameSource(TensorSource):
```
Comment (sandhawalia) [inline src/rbyte/io/video/ffmpeg_source.py:17]: Whats the rationale for appending `Source` ?

File: src/rbyte/io/video/vali_source.py [hunk 3]
```diff
@@ -29,8 +24,8 @@
 ]
 
 
-class ValiGpuFrameReader(FrameReader):
-    @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
+class ValiGpuFrameSource(TensorSource):
```
Comment (sandhawalia) [inline src/rbyte/io/video/vali_source.py:27]: Same here. Source is actually more or a `Reader`

File: src/rbyte/utils/functional.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    value: float | None = None,
+) -> Float[Tensor, "..."]:
+    _pad = [(0, 0) for _ in input.shape]
+    _pad[dim] = pad
+    _pad = list(mit.flatten(reversed(_pad)))
+
+    return F.pad(input, _pad, mode=mode, value=value)
+
+
+def pad_sequence(
```
Comment (sandhawalia) [inline src/rbyte/utils/functional.py:25]: https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pad_sequence.html

File: tests/data/zod/sequences/000002_short/camera_front_blur/000002_romeo_2022-06-13T10:50:07.092270Z.jpg
Comment (sandhawalia) [inline tests/data/zod/sequences/000002_short/camera_front_blur/000002_romeo_2022-06-13T10:50:07.092270Z.jpg:1]: Need to check if we have permissions for this

**Review by sandhawalia — APPROVED** [review]
Nothing glaring that stood out expcept. 

- Gonna check on ZOD license if we can add them to a public repo
- Some clarification around `Source` vs `Reader` suffix 
-

File: src/rbyte/io/video/ffmpeg_source.py (hunk 2 above)
Comment (egorchakov) [reply src/rbyte/io/video/ffmpeg_source.py:17]: not 100% sold on the naming myself but the rationale is that i updated the base protocol (`TensorSource`) to a simpler (and hopefully more pythonic) version:  
```
@runtime_checkable
class TensorSource(Protocol):
    def __getitem__(self, indexes: Iterable[Any]) -> Tensor: ...
    def __len__(self) -> int: ...
```

File: src/rbyte/utils/functional.py (hunk 4 above)
Comment (egorchakov) [reply src/rbyte/utils/functional.py:25]: was my first thought/attempt as well but need a slightly more flexible api: specifying which dim to pad
